pages/glossary/non-empty.md

```diff
@@ -3,15 +3,4 @@ title: Non-empty text string
 key: non-empty
 ---
 
-A string of characters (text) is considered *"non-empty"* if it contains 1 or more characters that are contained within any of the following unicode categories:
-
-- **Ll**: Letter, Lowercase
-- **Lu**: Letter, Uppercase
-- **Lt**: Letter, Titlecase
-- **Lo**: Letter, Other
-- **Lm**: Letter, Modifier
-- **Nd**: Number, Decimal Digit
-
-For more details on unicode categories, check out [www.fileformat.info/info/unicode/category/](http://www.fileformat.info/info/unicode/category/index.htm)
-
-  
\ No newline at end of file
+A string of characters is considered *non-empty* if the string contains characters that are not space characters.
```

Comment:
To ensure that space characters are perceived unambiguously perhaps you could use the 'Separator, Space' category of Unicode (and provide the link) similarly to what you do in the textual content definition.

Reply:
The HTML spec allows any character in IDs except ASCII whitespace:
https://html.spec.whatwg.org/multipage/dom.html#the-id-attribute
https://infra.spec.whatwg.org/#ascii-whitespace

So non-ASCII whitespace characters like these are valid in IDs
U+00A0 | NO-BREAK SPACE
U+200B | ZERO WIDTH SPACE

I think the shared definition is problematic because it's used in different contexts where the definition of space is different in each context:
- what is treated as space/empty by the HTML parser
- what is treated as space/empty and displayed as blank or zero width by fonts
- what is treated as space/empty and not voiced by assistive tech

Reply:
Agreed, "space characters" is insufficient. Please specify which characters are considered space characters.